### PR TITLE
Add multi-digit support for overlay badges and tests

### PR DESCRIPTION
**Pull request by langleyd**
Fix multi-digit notification badge in Windows taskbar overlay

Fix multi-digit overlay badges and add tests
The Windows taskbar overlay reused the favicon's pill expansion code, which pushed multi-digit counts off the 16x16 canvas (e.g. "10" showed as just "0"). Always draw a circle and scale the font down for multi-digit counts instead.

Native windows badges [only go to 99+](https://learn.microsoft.com/en-us/windows/apps/develop/notifications/badges) so we can mimic that.

Adds a Playwright test for favicon and overlay use cases.

### CODE REVIEW

File: apps/web/playwright/e2e/favicon/favicon.spec.ts [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+ *  1. Open a stub page with <link rel=icon> in real Chromium.
+ *  2. Bundle favicon.ts with esbuild and inject it into the page.
+ *  3. Set the link's href to the real Element favicon, then call new Favicon().badge(count).
+ *  4. The class rewrites the link's href asynchronously, so poll until it changes,
+ *     then read the badged PNG out as a data URI and write it to disk.
+ *  5. Display the PNG scaled-up in an <img> and snapshot it for regression diffs.
+ */
+
+import { test, expect } from "@playwright/test";
+import { build } from "esbuild";
```
Comment (t3chguy) [inline apps/web/playwright/e2e/favicon/favicon.spec.ts]: Seems really odd to be using `esbuild` here when it is actually built using `babel`, so the test is exercising something which may not match reality (edge case, sure).

This feels more like a unit test than a playwright test, we've used https://github.com/americanexpress/jest-image-snapshot in the past which works well enough, and in a vitest-future we can get this for free when using the browser mode.

File: apps/web/playwright/e2e/favicon/favicon.spec.ts [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+import path from "node:path";
+import fs from "node:fs/promises";
+import { fileURLToPath } from "node:url";
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+const FAVICON_SRC = path.resolve(__dirname, "../../../src/favicon.ts");
+const OUT_DIR = path.resolve(__dirname, "../../test-results/favicons");
```
Comment (t3chguy) [inline apps/web/playwright/e2e/favicon/favicon.spec.ts]: is it really a test *result*? feels more like something for a tmpdir

File: apps/web/playwright/e2e/favicon/favicon.spec.ts (hunk 1 above)
Comment (langleyd) [reply apps/web/playwright/e2e/favicon/favicon.spec.ts]: c05a3c2

File: vitest.config.ts [hunk 3]
```diff
@@ -46,6 +46,10 @@ export default defineConfig({
     test: {
         projects: [
             "{apps,modules,packages}/*/vitest.config.ts",
+            // Browser-mode tests for element-web. Kept as a separate project rather than
+            // enabling browser mode on apps/web/vitest.config.ts, so that only the handful
+            // of tests which need a real browser pay for one.
```
Comment (t3chguy) [inline vitest.config.ts]: And presumably the reason the config isn't a separate project within the single vitest config is the same as L53-L54

Which they have actually fixed but seemingly not released yet https://github.com/vitest-dev/vitest/pull/10846 - looks to be coming in v5

File: apps/web/vitest.config.ts [hunk 4]
```diff
@@ -49,6 +49,8 @@ export default defineProject({
     },
     test: {
         include: ["src/**/*.test.{ts,tsx}"],
+        // `*.browser.test.*` runs in a real browser instead — see vitest.browser.config.ts
+        exclude: [...defaultExclude, "src/**/*.browser.test.{ts,tsx}"],
```
Comment (t3chguy) [inline apps/web/vitest.config.ts]: Hierarchically this feels off `.browser-test`/`.test.browser` feels better but not going to fight it

File: apps/web/tsconfig.json [hunk 5]
```diff
@@ -16,7 +16,7 @@
         "jsx": "react",
         "lib": ["es2022", "es2024.promise", "dom", "dom.iterable", "ESNext.Collection"],
         "strict": true,
-        "types": ["node", "modernizr"],
+        "types": ["node", "modernizr", "@vitest/browser/matchers"],
```
Comment (t3chguy) [inline apps/web/tsconfig.json]: Shouldn't this only apply to the browser tests, otherwise you get runtime errors if you accidentally use the wrong matcher which your intellisense suggests

**Review by t3chguy — APPROVED** [review]
Looks great

File: apps/web/vitest.config.ts (hunk 4 above)
Comment (langleyd) [reply apps/web/vitest.config.ts]: aba57f287a1941041f1f9ffbd1b1ee861d6f9f5b

File: apps/web/tsconfig.json (hunk 5 above)
Comment (langleyd) [reply apps/web/tsconfig.json]: 6af5800e47ec052da9d8cb15df244cab4be9e320

**Comment by langleyd** [comment]
@t3chguy Do those additional changes seem ok to you?
